pages/onboarding/index.tsx

```diff
@@ -1,8 +1,60 @@
-import { Box, Button, Container, Typography } from '@mui/material';
+import { Box, Button, Container, TextField, Typography } from '@mui/material';
+import axios from 'axios';
+import jwt from 'jsonwebtoken';
 import { NextPage } from 'next';
 import Link from 'next/link';
+import { useRouter } from 'next/router';
+import React, { useState } from 'react';
+import Cookies from 'universal-cookie';
+
+import createAxiosInstance from 'client/lib/axios';
+import { UserProfile } from 'server/types/User';
+import { AUTH_COOKIE_NAME } from 'shared/constants';
+
+interface FormData {
+  bio: string;
+  twitter_username: string;
+  linkedin_url: string;
+  github_username: string;
+  website: string;
+}
+
+interface UseridData {
+  user: UserProfile;
+}
+
+const OnboardingPage: NextPage<UseridData> = ({ user }: UseridData) => {
+  const [formData, setFormData] = useState<FormData>({
+    bio: '',
+    twitter_username: '',
+    linkedin_url: '',
+    github_username: '',
+    website: '',
+  });
+
+  const router = useRouter();
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { name, value } = event.target;
+    setFormData(prevFormData => ({ ...prevFormData, [name]: value }));
+  };
+
+  const handleSubmit = async (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    try {
+      const response = await axios.patch(`/api/users/${user.id}`, formData);
+
+      if (response.status !== 200) {
+        throw new Error('Failed to update user profile');
+      }
```

Comment:
2xx status codes are technically okay, 4xx and above are errors.  Additionally, the `axios.patch` call above will error if the request failed, so this error is unreachable.